jerry-core/api/jerry-debugger.c

```diff
@@ -93,13 +93,18 @@ jerry_debugger_stop_at_breakpoint (bool enable_stop_at_breakpoint) /**< enable/d
  * Debugger server initialization. Must be called after jerry_init.
  */
 void
-jerry_debugger_init (uint16_t port) /**< server port number */
+jerry_debugger_init (jerry_debugger_transport_t *transport_p) /**< transport */
```

Comment:
Is this port specific definition part of the public API?

Reply:
Do you mean the jerry_debugger_transport_t type?  Yes, this is intended and part of the jerry_port_ api where each platform have to provide their own definition, hence why it was forward declared, so it can build.

Reply:
Unfortunately "it can build" is not a strong enough reason for something to be part of the public API. 

While I am ok with having a structure for communication, the public API part needs to be independent of everything: no jerry port dependency, no compiler define dependency, etc.

I think the first question is: do we want multiple transport layer support, or the recv/send/close could be provided by jerry port.

Reply:
@zherczeg When I originally started the the port, it was a simpler patch to just port the socket APIs, but I think @martijnthe really like to support multiple transports, so we kinds modified it to take a struct of handlers.  I mean we can submit a simpler patch to ports only socket calls and then a separate patch later to rework the APIs to support multiple transports, I think it's up to you and the JerryScript maintainers to make this kind of decision.  Honestly, my goal is to at least get the websockets working on Zephyr as part of this effort.  If we move the the struct definitions and APIs to from jerryscript-port to jerryscript-debugger.h, would that work for you?

Reply:
> Is this port specific definition part of the public API?

The *definition* should __NOT__ be part of the public API, only the forward declaration should be. That way there is no dependency. (Until one actually want to use the debugger feature, in which case one has to pick an implementation which would contain the definition of the struct as well.)

> I think the first question is: do we want multiple transport layer support

Yes, at least in our project we want to mix and match between 3 transports at run-time.

IMHO, a stronger reason for wanting a more dynamic transport interface as opposed to extending the "port surface" with static functions for the transport, is that static functions would make combining, composing and re-using transport implementations harder.

For example in our case, we want USB, BLE and a "pseudo transport" that just gathers the debug info (i.e. for printing human consumable exception traces). Imagine that someone already wrote each of these transports separately. Each transport would define the same set of functions/symbols. If I want to create a new transport that combines/chains/... them, I'd have to rename each function to avoid conflicts. Even if a simple "switch" between 2 transports would involve renaming each function.